### ISSUE TITLE
Add support for excluding a directory if a specified file exists

### ISSUE THREAD

**Issue by perillo** [WayneD/rsync#451]
[Restic](https://restic.readthedocs.io/en/latest/040_backup.html#excluding-files) supports the options `--exclude-if-present <name>` and  `--exclude-caches`. The latter internally uses `--exclude-if-present`,  using the [Cache Directory Tagging Specification](https://bford.info/cachedir/).

My use case is excluding `build` directories when building computer programs.

**Comment by idallen** [comment]
Yes, `rsync` would benefit from obeying the `CACHEDIR.TAG` that other programs obey, such as GNU `tar`. Saying that you can simulate the same thing by adding `rsync`-specific filter files everywhere isn't helpful since that (1) doubles the number of files one has to add everywhere, (2) opens the security risk of `--trust-sender`, and (3) only works for `rsync`.

This thread got duplicated at [#499][feature request: support Cache Directory Tagging Specification](https://github.com/RsyncProject/rsync/issues/499)
